web/source/kmwcallback.ts

```diff
@@ -915,8 +920,8 @@ namespace com.keyman {
           var _CacheableCommands = this._CacheCommands(Ldoc);
         }
       
-        var Lsel = Ldv.getSelection();
-        var LselectionStart = Lsel.focusNode.nodeValue ? Lsel.focusNode.substringData(0,Lsel.focusOffset)._kmwLength() : 0;  // I3319
+        var Lsel: Selection = Ldv.getSelection();
+        var LselectionStart = Lsel.focusNode.nodeValue ? (<Text> Lsel.focusNode).substringData(0,Lsel.focusOffset)._kmwLength() : 0;  // I3319
         
```

Comment:
Current edits in this file exist to aid analysis of existing code, which should facilitate before-and-after comparisons with the new classes.